core/src/main/java/io/questdb/griffin/engine/functions/catalogue/TableListFunctionFactory.java

```diff
@@ -228,41 +240,43 @@ public CharSequence getStrB(int col) {
 
                 @Override
                 public int getStrLen(int col) {
-                    return getStr(col).length();
+                    CharSequence str = getStr(col);
+                    return str != null ? str.length() : -1;
                 }
 
-                public boolean open(TableToken tableToken) {
+                private boolean open(TableToken tableToken) {
                     if (hideTelemetryTables && (Chars.equals(tableToken.getTableName(), TelemetryTask.TABLE_NAME)
                             || Chars.equals(tableToken.getTableName(), TelemetryConfigLogger.TELEMETRY_CONFIG_TABLE_NAME)
                             || Chars.startsWith(tableToken.getTableName(), sysTablePrefix))) {
                         return false;
                     }
 
-                    int pathLen = path.size();
-                    try {
-                        path.concat(tableToken).$();
-                        isSoftLink = Files.isSoftLink(path);
-                        path.concat(META_FILE_NAME).$();
-                        tableReaderMetadata.load(path);
-
-                        // Pre-read as much as possible to skip record instead of failing on column fetch
-                        tableId = tableReaderMetadata.getTableId();
-                        maxUncommittedRows = tableReaderMetadata.getMaxUncommittedRows();
-                        o3MaxLag = tableReaderMetadata.getO3MaxLag();
-                        partitionBy = tableReaderMetadata.getPartitionBy();
-                    } catch (CairoException e) {
-                        // perhaps this folder is not a table
-                        // remove it from the result set
-                        LOG.info()
-                                .$("cannot query table metadata [table=").$(tableToken)
-                                .$(", error=").$(e.getFlyweightMessage())
-                                .$(", errno=").$(e.getErrno())
-                                .I$();
-                        return false;
-                    } finally {
-                        path.trimTo(pathLen).$();
+                    if (getMetadata() == METADATA) {
+                        int pathLen = path.size();
+                        try {
+                            path.concat(tableToken).$();
+                            isSoftLink = Files.isSoftLink(path);
+                            path.concat(META_FILE_NAME).$();
+                            tableReaderMetadata.load(path);
```

Comment:
we should avoid reading file system from factories and leverage existing infrastructure, there is metadata pool available on the engine

Reply:
hey, this was a refactor, if the implementation underneath is not good I can fix it, but if it needs fixing it means it was broken. Was it broken?